libs/api/domains/air-discount-scheme/src/lib/flight-leg/flight-leg.service.ts

```diff
@@ -38,35 +41,81 @@ export class FlightLegService {
     )
   }
 
-  async getThisYearsFlightLegs(auth: User): Promise<FlightLeg[]> {
-    const getFlightsResponse = await this.getADSWithAuth(auth)
-      .privateFlightUserControllerGetUserFlights({
-        nationalId: auth.nationalId,
-      })
+  async getThisYearsUserAndRelationsFlightLegs(
+    auth: User,
+  ): Promise<FlightLeg[]> {
+    const flights = await this.getADSWithAuth(auth)
+      .privateFlightUserControllerGetUserAndRelationsFlights()
       .catch((e) => {
         this.handle4xx(e)
       })
 
-    if (!getFlightsResponse) {
+    if (!flights) {
       return []
     }
+
+    const relations: TUser[] = await this.getUserRelations(auth)
     const flightLegs: FlightLeg[] = []
 
-    getFlightsResponse.forEach((flight) => {
-      if (flight?.flightLegs) {
-        for (const flightLeg of flight.flightLegs) {
-          flightLegs.push({
-            ...flightLeg,
-            flight: {
-              bookingDate: flight.bookingDate,
-              id: flight.id,
-            },
-            travel: `${flightLeg.origin} - ${flightLeg.destination}`,
-          })
+    // The expected return value for the graphql layers has some extra properties
+    // We have to maintain circularity as well since the types are circular.
+    // Therefore we do some interesting object surgery
+    for (const flight of flights) {
+      // Not strictly needed but good practice for type safety
+      const relation = relations.find(
+        (relation) => relation.nationalId === flight.nationalId,
+      )
+      if (!relation) {
+        continue
+      }
+
+      // We construct new objects with the expected model properties
+      const constructedUser: FlightUser = {
+        ...relation,
+        name: `${relation.firstName} ${relation.lastName}`,
+      }
+      const constructedFlightLegs: FlightLeg[] = []
+
+      // UserInfo in flight.userInfo has gender as string in the generated schema
+      // but is a string union type, hence the `as Flight` coercion
+      const constructedFlight: Flight = {
+        ...flight,
+        user: constructedUser,
```

Comment:
Annoying to have to manually query and fill the user here. Could be included in original request or implemented as a field resolver with the dataloader pattern.

Reply:
Yeah, I agree.

I tried including this in the original response but didn't like the semantic changes to the user controller by querying flight details from there. 
Neither did I like the whole ADS backend becoming `forwardRef() =>` dependent upon introducing the UserModule to the FlightModule if I remember correctly.

Possibly I could do some code duplication but there is a hefty block of logic associated with getting a user with their calculated benefits.

Or just move this endpoint to the User Controller?